### DISCUSSION TITLE
Chemical potential

### DISCUSSION THREAD

**Discussion by jk262626** [nakib/elphbolt#139] [General]
Hi Nakib, thank you for the reply regarding chemical potential.

For more detail, I will describe my calculation. 
I'm calculating aluminum at 300K and the calculation results are different according to the chemical potential I used. 
In the first try, I used chemical potential calculated from band structure calculation (say 7. 95eV) and I got very large KO dev. value as attached file (slurm_300K_25).
In the second try, I used theoretical value estimated from 1D approach (say 11.6eV) and I got almost zero KO dev. value as attached file (slurm-30186358). 
I want to know how the chemical potential value can affect the simulation results. I thought only the relative amount of energy with regard to chemical potential is important. 
[Al_jk.zip](https://github.com/nakib/elphbolt/files/15223016/Al_jk.zip)

**Comment by nakib** [comment]
Dear jk262626,

The chemical potential (`chempot`) is used to dope a system. Typically one does not dope metals. And since you have a metal here, I am assuming that doping is not your intention. In that case, you want to set the chemical potential to the Fermi energy value that you get from your DFT calculation. Typically, you need to do an `nscf` calculation with a fine enough k-mesh to determine the Fermi energy. I am guessing you have already done this and that you have compared your electronic band structures from direct DFT and Wannier calculations. I can try to assist you further if you can send me these plots and the DFT value of the Fermi energy.

Another relevant set of quantities are the electronic reference energy (`enref`) and the Fermi surface thickness (`fsthick`). Please look up the documentation on these. In short, these two parameters are used to create a thin shell in the Brillouin zone. Since you are not going to dope the system, you want to have the same number for both `enref` and `chempot`.

I don't know what a 1D approach means in this context.

Best,
Nakib

**Reply by nakib** [reply]
Another comment: I noticed that you were repeating the phonon-phonon interactions calculation. This is a big waste of resources. Once the phonon-phonon vertices have been calculated, they can be reused for all other calculations that have the same q-mesh and structure.

**Reply by jk262626** [reply]
Thank you for the reply and comments!

1. Could you let me know where the electronic reference energy is used in the calculation? I tried to find the description of that, but couldn't find. 
2. Is the carrier concentration calculated with the following equation using DOS and temperature for the structure without doping? In this process, the chemical potential I put into isn't used. Am I right?
<img width="164" alt="Screenshot 2024-05-08 at 9 29 24 PM" src="https://github.com/nakib/elphbolt/assets/138066037/41f833c5-f872-4e8f-b313-a25ae3f7374b">

3. Then, how is the carrier density calculated for the structure with doping using chemical potentai?

**Reply by nakib** [reply]
1. https://github.com/nakib/elphbolt?tab=readme-ov-file#electrons
2. `src/electron.f90` -> `subroutine calculate_carrier_conc`
3. `src/electron.f90` -> `subroutine calculate_chempot`

The carrier concentration is only needed for electron-charged impurity scattering and conversion of conductivity to mobility. Since mobility is typically reported only for non-metallic systems, for metals I don't compute the carrier concentration at all.

Best,
Nakib

**Reply by jk262626** [reply]
Thank you for the answer. I think I'm using the wrong index number for indlowband and indhighband for Al. 
How can I determine the lowest and highest transport band index for metal from the band structure? 
It's quite not obvious. Is that like finding a band that has the lowest energy around the Fermi level?

**Reply by nakib** [reply]
This becomes clear when you plot the DFT and Wannierized bands on top of each other. You have a certain number of Wannier bands. Only a few of these live inside the shell defined by the `fsthick` and `enref`. You need to provide the lowest and highest Wannier band indices for this subset.

Best,
Nakib

**Reply by jk262626** [reply]
Oh I see. For example like the figure below, is it summing over the bands and k points that live inside fermi level thickness (fsthick) while going through from the lowest to the highest transport band index I set? If that's right, theoretically I just can put 1 to 6 (if the total Wannier band number is 6) but it takes longer computation time, and so it's better to select the band that is crossing over Fermi level. 

![image](https://github.com/nakib/elphbolt/assets/138066037/1e73d6a4-254f-42c4-9303-530a3e60162c)

**Reply by nakib** [reply]
Yes, that's correct.
